### PR TITLE
DeltavisionReader.java: complete objectives metadata from softWoRx 7.0.0

### PR DESCRIPTION
**Pull request by carandraug**
Following an unrelated discussion on issue #2656 , I took a look at the list of objective ids for missing values.  This commit completes the list with the data from the latest version of softWoRx (version 7.0.0).

This commit also adds the name of the objective in front of the id as a comment which hopefully helps in finding issues in filling the correction type.  I was also confused about the calibratedmagnification, I'm not sure how to know if the magnification value is calibrated or not from the data on that file.

### CODE REVIEW

File: components/formats-gpl/src/loci/formats/in/DeltavisionReader.java [hunk 1]
```diff
@@ -1451,863 +1451,1321 @@ else if (lensID < 20000) {
...
         lensNA = 0.40;
+        magnification = 10.0;
+        workingDistance = 3.1;
+        immersion = getImmersion("Air");
+        model = "1-U2B823";
+        break;
+      case 10116: // Olympus 4X/0.16, U-Plan S-Apo, UIS2, 1-U2B822
+        lensNA = 0.16;
+        magnification = 4.0;
+        workingDistance = 13;
```
Comment (sbesson) [inline components/formats-gpl/src/loci/formats/in/DeltavisionReader.java]: This fails Travis with

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.7.0:compile (default-compile) on project formats-gpl: Compilation failure
[ERROR] /home/travis/build/openmicroscopy/bioformats/components/formats-gpl/src/loci/formats/in/DeltavisionReader.java:[1573,27] incompatible types: int cannot be converted to java.lang.Double
[ERROR] -> [Help 1]
```

Can you try `workingDistance = 13.0;` ?

**Comment by carandraug** [comment]
Sorry. I should have tried to rebuild it myself first. It is fixed now (at least it builds)

For what is worth, the whole switch/case blocks were generated automatically with this perl script https://github.com/MicronOxford/scripts/blob/master/auto/dvlenses2readercases.pl

The script has an issue with figuring out the right correction type. It tries a series of regexs but they are not always in the same order so sometimes picks UV and other times PlanApo. I could fix it by forcing a order on the regexp but I think the issue here is that I don't even know myself which correction to use sometimes. For example, for "Olympus 100X/1.30, D Plan Apo UV" should it be PlanApo or UV?

File: components/formats-gpl/src/loci/formats/in/DeltavisionReader.java [hunk 2]
```diff
@@ -1451,863 +1451,1321 @@ else if (lensID < 20000) {
...
-        correction = getCorrection("Fluar");
         model = "44 02 55 (01)";
+        correction = getCorrection("Fluar");
         break;
-      case 14402:
+      case 14402: // Zeiss 40X/1.00, Plan - APOCHROMAT, Phase3
         lensNA = 1.00;
+        magnification = 40.0;
         immersion = getImmersion("Oil");
+        model = "44 07 51 ";
```
[the remark this answers is not in the document]
Comment (carandraug) [reply components/formats-gpl/src/loci/formats/in/DeltavisionReader.java]: Not really. I should have trimmed the values when I parsed it from the data files we get from softworx. Now fixed in fdc0a27e607377

**Comment by carandraug** [comment]
>  Unless the calibrated magnification is expected to be stored elsewhere, a follow-up might involve the cleanup of the calibratedMagnification handling.

With commit 735eacef819055 I removed the whole `calibratedMagnification` stuff.

**Pull request merged**